scheduler/distributed_task_test.go

```diff
@@ -265,39 +270,109 @@ func TestDistributedSubscriptions(t *testing.T) {
 			schTask.RemoteManagers.Add(fmt.Sprintf("127.0.0.1:%v", port1), remoteMockManager)
 
 			// Start task. We expect it to fail while subscribing deps
-			terrs := sch.StartTask(t.ID())
+			terrs := s.StartTask(t.ID())
 			So(terrs, ShouldNotBeNil)
 			Convey("So dependencies should have been unsubscribed", func() {
 				// Ensure that unsubscribe call count is equal to subscribe call count
 				// i.e that every subscribe call was followed by an unsubscribe since
 				// we errored
 				So(remoteMockManager.UnsubscribeCallCount, ShouldEqual, remoteMockManager.SubscribeCallCount)
-				So(localMockManager.UnsubscribeCallCount, ShouldEqual, localMockManager.UnsubscribeCallCount)
+				So(localMockManager.UnsubscribeCallCount, ShouldEqual, localMockManager.SubscribeCallCount)
 			})
 		})
-
 		Convey("Starting task should succeed if all deps are available", func() {
-			//Create a task
-			//Create a workflowmap
-			wf := dsWFMap(port1)
-			// Create a task that is not started immediately so we can
-			// validate deps correctly.
-			t, errs := sch.CreateTask(schedule.NewSimpleSchedule(time.Second), wf, false)
-			So(len(errs.Errors()), ShouldEqual, 0)
-			So(t, ShouldNotBeNil)
-			schTask := t.(*task)
-			localMockManager := &subscriptionManager{Fail: false}
-			schTask.RemoteManagers.Add("", localMockManager)
-			remoteMockManager := &subscriptionManager{Fail: false}
-			schTask.RemoteManagers.Add(fmt.Sprintf("127.0.0.1:%v", port1), remoteMockManager)
-			terrs := sch.StartTask(t.ID())
-			So(terrs, ShouldBeNil)
-			Convey("So all depndencies should have been subscribed to", func() {
-				// Ensure that unsubscribe call count is equal to subscribe call count
-				// i.e that every subscribe call was followed by an unsubscribe since
-				// we errored
-				So(localMockManager.SubscribeCallCount, ShouldBeGreaterThan, 0)
-				So(remoteMockManager.SubscribeCallCount, ShouldBeGreaterThan, 0)
```

Comment:
Lines 285-301 moved to convey "Task is expected to run until being stopped" in [line 284](https://github.com/intelsdi-x/snap/pull/1548/files#diff-cac84587e02930df0708d14acc453f20R284)